openmdao/core/system.py

```diff
@@ -1615,6 +1616,53 @@ def _init_relevance(self, mode):
         if self._use_derivatives:
             desvars = self.get_design_vars(recurse=True, get_sizes=False, use_prom_ivc=False)
             responses = self.get_responses(recurse=True, get_sizes=False, use_prom_ivc=False)
+
+            # If you have response aliases, check for overlapping indices.
+            aliases = {key: value for key, value in responses.items() if value['path']}
+            if len(aliases) > 0:
+
+                used_idx = {}
+                discrete = self._var_allprocs_discrete
+
+                for name in aliases:
+
+                    if name in discrete['input'] or name in discrete['output']:
+                        continue
+
+                    path = responses[name]['path']
+                    if path is None:
+                        path = name
+
+                    if name not in used_idx and path not in used_idx:
+                        size = self._var_allprocs_abs2meta['output'][path]['global_size']
+                        used_idx[path] = np.zeros(size, dtype=int)
+
+                        # Source won't be in aliases, but we need its indices if a constraint is
+                        # declared without an alias.
+                        if path in responses:
+                            idx = responses[path].get('indices')
+
+                            if idx is None:
+                                used_idx[path][:] += 1
+                            else:
+                                used_idx[path][idx.flat()] += 1
+                        else:
+                            # If an alias is in responses, but the path isn't, then we need to
+                            # make sure the path is present for the relevancy calculation.
+                            responses[path] = responses[name]
+
+                    indices = responses[name].get('indices')
+                    if indices is None:
+                        used_idx[path][:] += 1
+                    else:
+                        used_idx[path][indices.flat()] += 1
+
+                    overlap = np.where(used_idx[path] > 1)
```

Comment:
Since you're not using the actual overlapping indices for anything, you could instead just use `overlap = np.any(used_idx[path] > 1)`

Reply:
Replaced the code.